packages/payment/src/services/payment-module.ts

```diff
@@ -201,44 +201,67 @@ export default class PaymentModuleService<
     )
   }
 
-  @InjectTransactionManager("baseRepository_")
+  @InjectManager("baseRepository_")
   async createPaymentSession(
     paymentCollectionId: string,
-    data: CreatePaymentSessionDTO,
+    input: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    const created = await this.paymentSessionService_.create(
-      {
-        provider_id: data.provider_id,
-        amount: data.providerContext.amount,
-        currency_code: data.providerContext.currency_code,
-        payment_collection: paymentCollectionId,
-      },
-      sharedContext
-    )
+    let paymentSession: PaymentSession
 
     try {
-      const sessionData = await this.paymentProviderService_.createSession(
-        data.provider_id,
-        {
-          ...data.providerContext,
-          resource_id: created.id,
-        }
-      )
+      const providerSessionSession =
+        await this.paymentProviderService_.createSession(input.provider_id, {
+          context: input.context ?? {},
+          amount: input.amount,
+          currency_code: input.currency_code,
+        })
+
+      input.data = {
+        ...input.data,
+        ...providerSessionSession,
+      }
 
-      await this.paymentSessionService_.update(
-        {
-          id: created.id,
-          data: sessionData,
-        },
+      paymentSession = await this.createPaymentSession_(
+        paymentCollectionId,
+        input,
         sharedContext
       )
-
-      return await this.baseRepository_.serialize(created, { populate: true })
-    } catch (e) {
-      await this.paymentSessionService_.delete([created.id], sharedContext)
-      throw e
+    } catch (error) {
+      // In case the session is created at the provider, but fails to be created in Medusa,
```

Comment:
Note that this is not enough if we want to always clean up, since what can happen is:
1. Create session in provider
2. Your node crashes at any point between the session being created, and `deleteSession` being executed

So you'll end up with an orphaned session. I guess in this case it doesn't matter as sessions expire I assume.

With that in mind, do we even need this cleanup step? Failures will be rare on our side anyway, so not sure if we need the added complexity and false security that it will fully clean up.

Reply:
Orphaned sessions do not really matter, no. However, I don't find this to add much complexity to the operation, so I'll keep this in for now. I am open to revisiting it in the future.